src/libs/routing/utils.ts

```diff
@@ -8,7 +8,7 @@ function toValue(mix: string | undefined) {
   if (str === 'true') return true;
   if (str.startsWith('0x')) return str;
   if (str.startsWith('0X')) return str;
-  return +str * 0 === 0 ? +str : str;
+  return +str * 0 === 0 && +str + '' === str ? +str : str;
```

Comment:
I agree with Francois and not sure why it was resolved - this line is cryptic - please move it to a function called e.g. `convertIfNumeric`

Reply:
as discussed in the group chat, this is whole parser function is copy paste from tanstack router library. Francois is removing the whole thing in his PR that addresses the duplicate strategy issue.